lib/src/rules/no_implicit_dynamic.dart

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.annotate_types;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:linter/src/linter.dart';
+import 'package:linter/src/util.dart';
+
+const desc = 'Implicit use of dynamic.';
+
+const details = '''
+**AVOID** using "implicitly dynamic" values.
+
+Untyped / dynamic invocations may fail or be slower at runtime, but dynamic
+types often creep up unintentionally. Explicitly mark variables or return types
+as `dynamic` (instead of `var`) to express your intent unequivocally.
+
+Note: this works best with the --strong command-line flag and after disabling
+both `always_specify_types` and `always_declare_return_types` lints.
+
+**GOOD:**
+```dart
+String trim(String s) => s.trim();
+
+main() {
+  var s = trim(' a ').toUpperCase();
+
+  dynamic x;
+  x = ... ;
+  x.reallyNotSureThisExists();
+}
+```
+
+**BAD:**
+```dart
+trim(s) => s.trim();
+
+main() {
+  var s = trim(1).toUpperCase();
+
+  var x;
+  x = ... ;
+  x.reallyNotSureThisExists();
+}
+```
+''';
+
+class NoImplicitDynamic extends LintRule {
+  NoImplicitDynamic()
+      : super(
+            name: 'no_implicit_dynamic',
+            description: desc,
+            details: details,
+            group: Group.style);
+
+  @override
+  AstVisitor getVisitor() => new Visitor(this);
+}
+
+// TODO(ochafik): Handle implicit return types of method declarations (vs. overrides).
+class Visitor extends SimpleAstVisitor {
+  final LintRule rule;
+  Visitor(this.rule);
+
+  Element _getBestElement(Expression node) {
```

Comment:
Consider to use getCanonicalElementFromIdentifier method from DartTypeUtilities (if applies)